Dynamic Form Docs Section Update (#10734)
RasaHQ/rasa: docs/docs/forms.mdx

        return additional_slots + domain_slots
```

> If conversely, you'd prefer to remove a slot from the form's `required_slots` defined in the domain file in certain conditions,

Reviewer: ```suggestion
If conversely, you want to remove a slot from the form's `required_slots` defined in the domain file under certain conditions,
```

> we recommend to copy the `domain_slots` over to a new variable and applying changes to that new variable instead of modifying directly

Reviewer: ```suggestion
you can copy the `domain_slots` over to a new variable and apply changes to that new variable instead of directly modifying
```

Reviewer: will anything bad happen if they do modify it directly or is it just for clarity?

Author: They won't get the behaviour they want (not requesting the removed slot), because Rasa will instead use the list of required slots defined in the domain, I explained how this occurs in the comment [here](https://github.com/RasaHQ/rasa/issues/10536#issuecomment-1020405414).

Reviewer: Ah I see thanks! 
In that case we should be stronger in the wording e.g.
```
you should copy the `domain_slots` over to a new variable and apply changes to that new variable instead of directly modifying `domain_slots`. Directly modifying `domain_slots` can cause unexpected behaviour.
```
